testing infra and some fixes

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #162183

This PR is quite large in that it covers most of rough edges in the new strict export flow:

1. Handle nn_module_stack correctly now that we are tracing wrapper module
2. module_call_spec needs to get queried from source directly because we are not running the bytecode anymore.
3. Correct input and output handling.

@diff-train-skip-merge

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

tugsbayasgalan committed:
testing infra and some fixes

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "testing infra and some fixes"




[ghstack-poisoned]

tugsbayasgalan committed:
Update on "testing infra and some fixes"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "testing infra and some fixes"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "testing infra and some fixes"


This PR is quite large in that it covers most of rough edges in the new strict export flow:

Handle nn_module_stack correctly now that we are tracing wrapper module
module_call_spec needs to get queried from source directly because we are not running the bytecode anymore.
Correct input and output handling.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81793200).

tugsbayasgalan committed:
Update on "testing infra and some fixes"


This PR is quite large in that it covers most of rough edges in the new strict export flow:

1. Handle nn_module_stack correctly now that we are tracing wrapper module
2. module_call_spec needs to get queried from source directly because we are not running the bytecode anymore.
3. Correct input and output handling.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

Differential Revision: [D81793200](https://our.internmc.facebook.com/intern/diff/D81793200)

[ghstack-poisoned]

tugsbayasgalan commented:
@pytorchbot merge -f "Landed internally"

huydhn commented:
@pytorchbot merge

huydhn commented:
@pytorchbot merge -f 'Merge internally'

huydhn commented:
@pytorchbot revert -m 'Failing a test on macos' -c nosignal

export/test_strict_export_v2.py::StrictExportV2TestExport::test_vmap_strict_export_v2 [GH job link](https://github.com/pytorch/pytorch/actions/runs/17570111947/job/49904829863) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/d8b6622bb6a3879d3832ab6cdc26ff4188ea4a2d)

You also want to rebase this after I reland https://github.com/pytorch/pytorch/pull/162170 (merge conflict on aot_compile.py)

huydhn commented:
> @pytorchbot merge -i
> 
> (Initiating merge automatically since Phabricator Diff has merged, merging with -i because oss signals were bypassed internally)

Now where did you come from?  The internal diff has been reverted

tugsbayasgalan committed:
Update on "testing infra and some fixes"


This PR is quite large in that it covers most of rough edges in the new strict export flow:

1. Handle nn_module_stack correctly now that we are tracing wrapper module
2. module_call_spec needs to get queried from source directly because we are not running the bytecode anymore.
3. Correct input and output handling.

diff-train-skip-merge

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "testing infra and some fixes"


This PR is quite large in that it covers most of rough edges in the new strict export flow:

1. Handle nn_module_stack correctly now that we are tracing wrapper module
2. module_call_spec needs to get queried from source directly because we are not running the bytecode anymore.
3. Correct input and output handling.

diff-train-skip-merge

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

tugsbayasgalan commented on torch/export/_unlift.py:
```diff
@@ -729,14 +729,35 @@ def _unlift_exported_program_lifted_states(
         )
+
+        # TODO (tmanlaibaatar)
+        # This is band-aid solution to export new tracer replacing
```
cc: @avikchaudhuri  I guess we need to fix this before Autoparallel needing to run the guards in a different way or if they use lower level export APIs.

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82039285).

tugsbayasgalan commented:
@pytorchbot rebase

tugsbayasgalan committed:
Update on "testing infra and some fixes"


This PR is quite large in that it covers most of rough edges in the new strict export flow:

1. Handle nn_module_stack correctly now that we are tracing wrapper module
2. module_call_spec needs to get queried from source directly because we are not running the bytecode anymore.
3. Correct input and output handling.

diff-train-skip-merge

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

Differential Revision: [D82039285](https://our.internmc.facebook.com/intern/diff/D82039285)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82039285).

tugsbayasgalan committed:
Update on "testing infra and some fixes"


This PR is quite large in that it covers most of rough edges in the new strict export flow:

1. Handle nn_module_stack correctly now that we are tracing wrapper module
2. module_call_spec needs to get queried from source directly because we are not running the bytecode anymore.
3. Correct input and output handling.

diff-train-skip-merge

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

Differential Revision: [D82039285](https://our.internmc.facebook.com/intern/diff/D82039285)

[ghstack-poisoned]

tugsbayasgalan commented:
@pytorchbot merge -f "Landed internally"

huydhn commented:
@pytorchbot merge -f 'This has been landed internally'
